src/main/java/com/sap/oss/phosphor/fosstars/data/artifact/ReleaseInfoFromMaven.java

```diff
@@ -61,10 +63,25 @@ private JsonNode mavenArtifactReleaseInfo(MavenArtifact mavenArtifact) throws IO
         }
       }
       values.update(RELEASED_ARTIFACT_VERSIONS.value(new ArtifactVersions(artifactVersions)));
+      updateArtifactVersion(mavenArtifact, artifactVersions, values);
       return this;
     }
 
     values.update(RELEASED_ARTIFACT_VERSIONS.unknown());
+    values.update(ARTIFACT_VERSION.unknown());
     return this;
   }
+
+  private void updateArtifactVersion(MavenArtifact mavenArtifact,
+      Set<ArtifactVersion> artifactVersions, ValueSet values) {
+
+    mavenArtifact.version()
+        .flatMap(version -> artifactVersions.stream()
+          .filter(v -> v.getVersion().equals(version))
+          .findFirst())
+        .ifPresent(match ->
+            values.update(
+                ARTIFACT_VERSION.value(
+                    new ArtifactVersion(match.getVersion(), match.getReleaseDate()))));
```

Comment:
```suggestion
        .ifPresent(match ->
            values.update(
                ARTIFACT_VERSION.value(match)));
```
Would this not work?

Reply:
Yes and Thanks